lib/shopify_api/only_sync_dirty_attributes.rb

```diff
@@ -0,0 +1,43 @@
+module ShopifyAPI
+  module OnlySyncDirtyAttributes
+    def method_missing(method_symbol, *args)
+      method_name = method_symbol.to_s
+
+      if match_data = /(?<attribute_name>.+)(?:=)$/.match(method_name)
+        changed << match_data[:attribute_name]
+      end
+
+      super
+    end
+
+    def save(*)
+      with_dirty_attributes { super }
+    end
+
+    private
+
+    def with_dirty_attributes
+      old_attributes = @attributes.dup
+      slice_attributes_on_changed
+      yield
+      clear_changed_attributes
+    rescue ActiveResource::ClientError,
+      ActiveResource::ConnectionError,
+      ActiveResource::TimeoutError => e
```

Comment:
`=> e` is not necessary. Also, if you just re-raise the stack trace will be preserved properly